src/main/java/com/epam/prejap/teatrees/game/Grid.java

```diff
@@ -0,0 +1,149 @@
+package com.epam.prejap.teatrees.game;
+
+import com.epam.prejap.teatrees.block.Block;
+
+import java.util.Arrays;
+import java.util.Objects;
+
+/**
+ * Provides Playfield an api for working with 2D array.
+ * @author Dominik Kaminski
+ * @author Przemyslaw Szewczyk
+ */
+final class Grid {
+    private final byte[][] grid;
+    private final int rows;
+    private final int cols;
+
+    private Grid(byte[][] grid, int rows, int cols) {
+        this.grid = grid;
+        this.rows = rows;
+        this.cols = cols;
+    }
+
+    Grid(int rows, int cols) {
+        this(new byte[rows][cols], rows, cols);
+    }
+
+    Grid(byte[][] grid) {
+        this(grid, grid.length, grid[0].length);
+    }
+
+    int cellAt(int row, int col) {
+        return grid[row][col];
+    }
+
+    /**
+     * It works like this - firstly it finds the first row that has only zeros and sets it
+     * as a bound. Secondly it counts how many times line removal occurred. Finally, it creates
+     * new lines below the bound. Obviously number of created lines is equal to the number of
+     * removed lines.
+     */
+    void removeCompleteLines() {
+        int[] fallingRows = new int[rows];
+        int bound = getWorkAreaBound(fallingRows);
+        int counter = countRemovedLines(fallingRows, bound);
+        fillEmptySpace(bound, counter);
+    }
+
+    private void fillEmptySpace(int bound, int counter) {
+        while (counter > 0) {
+            grid[bound + counter - 1] = new byte[cols];
+            counter--;
+        }
+    }
+
+    private int countRemovedLines(int[] fallingRows, int bound) {
+        int counter = 0;
+        for (int i = rows - 1; i >= bound; i--) {
+            if (fallingRows[i] == 0) counter++;
+            else grid[i + counter] = grid[i];
+        }
+        return counter;
+    }
+
+    private int getWorkAreaBound(int[] fallingRows) {
```

Comment:
This method is fairly complex. And yet, it's private so there's not test for it.

Reply:
Method removed.